pkg/cri/server/container_stop.go

```diff
@@ -129,9 +130,12 @@ func (c *criService) stopContainer(ctx context.Context, container containerstore
 		if err != nil {
 			return errors.Wrapf(err, "failed to parse stop signal %q", stopSignal)
 		}
-		log.G(ctx).Infof("Stop container %q with signal %v", id, sig)
-		if err = task.Kill(ctx, sig); err != nil && !errdefs.IsNotFound(err) {
-			return errors.Wrapf(err, "failed to stop container %q", id)
+
+		if atomic.CompareAndSwapUint32(&container.IsSignaled, 0, 1) {
```

Comment:
I think `Container` is copied from `containerStore.Get()` and the atomic call with `&container.IsSignaled` doesn't change the origin data.

And it doesn't work if containerd restarts.

Reply:
Thanks, updated.

https://github.com/kubernetes/kubernetes/pull/98507#discussion_r571285790
The container restart is acceptable.

Reply:
@fuweid 
Is it possible to add parameters to CreateContainerRequest to configure to send the signal only once